ketherhomepage/static.go

```diff
@@ -192,34 +192,72 @@ func (w *KetherWatcher) Watch(duration time.Duration) {
 		data := KetherData{BlockNumber: int(blockNumber.Int64()), Ads: ads}
 		json, err := json.Marshal(data)
 		if err != nil {
-			log.Printf("%s: Couldn't marshal ads to json: %v", w.name, err)
+			log.Errorf("%s: Couldn't marshal ads to json: %v", w.name, err)
 			continue
 		}
 
+		log.Printf("%s: Writing JSON", w.name)
 		jsonW := w.jsonObject.NewWriter(w.ctx)
-		jsonW.Write(json)
-		jsonW.Close()
-		log.Printf("%s: Wrote JSON", w.name)
+		_, err = jsonW.Write(json)
```

Comment:
Optional nit: I usually write these as 

```go
if _, err := jsonW.Write(json); err != nil {
    ...
}
```

Primarily to avoid overriding/polluting the outer-scope `err`.

Reply:
ah this is exactly what i was asking about earlier

Reply:
What you were asking about earlier still stands though, since you can have another err in the outer scope and it can be confusing afterwards about which err you're dealing with (it's not ambiguous in the language or anything, but a mistake that one could make nonetheless).